src/components/WidgetLock/index.tsx

```diff
@@ -0,0 +1,111 @@
+import {
+    Button,
+    Card,
+    CardBody,
+    CardSubtitle,
+    CardText,
+    CardTitle,
+    Col,
+    Container,
+    Popover,
+    PopoverBody,
+    PopoverHeader,
+    Row,
+    Tooltip,
+    UncontrolledTooltip,
+} from "reactstrap";
+import cx from "classnames";
+import styles from "./WidgetLock.module.scss";
+import { LockTypeEnum } from "types/widget-locks";
+import { CSSProperties, useEffect, useRef, useState } from "react";
+import "./WidgetLock.module.scss";
+interface Props {
+    size?: "small" | "normal" | "large";
+    lockReason?: string;
+    lockDescription?: string;
+    lockType?: LockTypeEnum;
+}
+
+// TODO: Add a refresh button
+const WidgetLock = ({
+    size = "normal",
+    lockReason = "There was some problem",
+    lockDescription = "There was some issue please try again later",
+    lockType = LockTypeEnum.GENERIC,
+}: Props) => {
+    const iconClasses = cx(
+        `fa ${styles.lockWidgetIcon}`,
+        { [styles.normal]: size === "normal" },
+        { [styles.small]: size === "small" },
+        { [styles.large]: size === "large" },
+        { "fa-triangle-exclamation": lockType === LockTypeEnum.GENERIC },
+        { "fa-house-laptop": lockType === LockTypeEnum.CLIENT },
+        { "fa-lock": lockType === LockTypeEnum.PERMISSION },
+        { "fa-wifi": lockType === LockTypeEnum.NETWORK },
+        { "fa-database": lockType === LockTypeEnum.SERVER_ERROR || lockType === LockTypeEnum.SERVICE_ERROR },
+    );
+    const [popoverOpen, setPopoverOpen] = useState(false);
+    const popoverRef = useRef<HTMLDivElement | null>(null);
+
+    const togglePopover = () => setPopoverOpen(!popoverOpen);
+
+    useEffect(() => {
+        const handleClickOutside = (event: MouseEvent) => {
+            if (popoverRef.current && !popoverRef.current.contains(event.target as Node)) {
+                setPopoverOpen(false);
+            }
+        };
+
+        document.addEventListener("mousedown", handleClickOutside);
```

Comment:
or you could use https://reactstrap.github.io/?path=/docs/components-popover--popover#focus